src/net.h

```diff
@@ -268,8 +269,7 @@ class CConnman
         }
     };
 
-    template<typename Callable>
-    void ForEachNode(Callable&& func) const
+    void ForEachNode(NodeFn func) const
```

Comment:
As the func is not consumed, shouldn't this keep the uref `&&`?

Reply:
This has to take a std::function argument so the lambda object is converted to a std::function and the EXCLUSIVE_LOCKS_REQUIRED requirement can by bypassed, otherwise static analysis will not accept this

Reply:
```suggestion
    void ForEachNode(const NodeFn& func) const
```

should work and be preferable? Obviously a style nit, as it doesn't make a difference with the current code.

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/19979#issuecomment-695243985).